subsys/bluetooth/controller/rng_driver.c

```diff
@@ -31,11 +33,18 @@ static int rng_driver_get_entropy(struct device *dev, u8_t *buf, u16_t len)
 	u8_t *p_dst = buf;
 	u16_t bytes_left = len;
 	while (bytes_left > 0) {
-		int32_t bytes_read;
-		while ((bytes_read = ble_controller_rand_vector_get(p_dst, bytes_left)) <= 0) {
-			/* Put the thread on wait until next interrupt to get more
-			 * random values. */
-			k_sem_take(&rng_data.sem_sync, K_FOREVER);
+		int32_t bytes_read = 0;
+		while (bytes_read <= 0) {
+			if (!MULTITHREADING_LOCK_ACQUIRE()) {
```

Comment:
One thing I'm concerned about is that a thread, which called this function, can be blocked forever if it failed to acquire the lock and semaphore is nevere given (beacuse IRQ is disabled). So the thread will stuck in k_sem_take().

How about this code:

> while (MULTITHREADING_LOCK_ACQUIRE()) {
>    k_yield();
> }
> bytes_read = ble_controller_rand_vector_get(p_dst, bytes_left);
> MULTITHREADING_LOCK_RELEASE();
> k_sem_take(&rng_data.sem_sync, K_FOREVER);

Reply:
Not really. In case of FOREVE/Timeout blocking,`k_sem_take()` (which is called by the acquire function) will automatically put the calling thread in pending mode if semaphore cannot be taken.

Reply:
After offline discussion it was decided to add one more function called MULTITHREADING_LOCK_ACQUIRE_FOREVER() to be used inside the RNG driver, because of the implementation of ble_controller and how this API is used by applications.

So this function will be used here, in rng_driver_entropy_get and in rng_driver_entropy_get_isr() for busywait case.